polling/test_service.py

```diff
@@ -1,14 +1,9 @@
 class TestService:
     def __init__(self):
-        self.try_attempts = 0
         self.error_attempts = 5
 
-    async def get_service_result(self, input):
-        print(
-            f"Attempt {self.try_attempts}"
-            f" of {self.error_attempts} to invoke service"
-        )
-        self.try_attempts += 1
-        if self.try_attempts % self.error_attempts == 0:
+    async def get_service_result(self, input, attempt: int):
+        print(f"Attempt {attempt}" f" of {self.error_attempts} to invoke service")
+        if attempt % self.error_attempts == 0:
```

Comment:
This is an unusual condition for a `return` -- it shouldn't be `attempt == self.error_attempts`?